inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -212,32 +212,18 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     bool workModeAuto = workMode != fullConfig.end() && workMode->second == InferenceEngine::PluginConfigParams::YES;
     auto priorities = fullConfig.find(MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES);
 
-    // not found device priorities for -d AUTO use case
+    // if workMode is AUTO
+    if (workModeAuto) {
+        CheckConfig(fullConfig);
+        return std::make_shared<MultiDeviceExecutableNetwork>(modelPath, network, fullConfig, this);
+    }
+
     if (priorities == fullConfig.end()) {
-        if (workModeAuto) {
-            std::string allDevices;
-            auto availableDevices = GetCore()->GetAvailableDevices();
-            if (availableDevices.empty()) {
-                IE_THROW(NotFound) << "No available device found";
-            }
-            for (auto&& device : availableDevices) {
-                allDevices += device;
-                allDevices += ((device == availableDevices[availableDevices.size()-1]) ? "" : ",");
-            }
-            metaDevices = ParseMetaDevices(allDevices, fullConfig);
-            multiNetworkConfig.insert({MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, allDevices});
-        } else {
-            IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for " << GetName() << " device";
-        }
+        IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for " << GetName() << " device";
     } else {  // for use case -d MULTI:xPU or -d AUTO:xPU
         metaDevices = ParseMetaDevices(priorities->second, fullConfig);
         multiNetworkConfig.insert(*priorities);
     }
-    // check if it is -d AUTO or -d AUTO:xPU use case
-    if (workModeAuto) {
-        auto targetDevice = SelectDevice(metaDevices, networkPrecision);
-        metaDevices = { targetDevice };
-    }
```

Comment:
all the code below with vector of 'loads' executor and etc somewhat duplicates your new code that alos does async loads. the primary diff is 
- the code below does wait for all devices (just like legacy MULTI should do)
- the code below has no futures, etc

Is it possible to refactor/merge these 2 code paths (your new and the one below)? or it would be too much changes?